homeassistant/components/apple_tv/config_flow.py

```diff
@@ -0,0 +1,449 @@
+"""Config flow for Apple TV integration."""
+from ipaddress import ip_address
+import logging
+from random import randrange
+
+from pyatv import const, convert, exceptions, pair, scan
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_NAME,
+    CONF_PIN,
+    CONF_PROTOCOL,
+    CONF_TYPE,
+)
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    CONF_CREDENTIALS,
+    CONF_CREDENTIALS_AIRPLAY,
+    CONF_CREDENTIALS_DMAP,
+    CONF_CREDENTIALS_MRP,
+    CONF_IDENTIFIER,
+    CONF_START_OFF,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+INPUT_PIN_SCHEMA = vol.Schema({vol.Required(CONF_PIN, default=None): int})
+
+DEFAULT_START_OFF = False
+PROTOCOL_PRIORITY = [const.Protocol.MRP, const.Protocol.DMAP, const.Protocol.AirPlay]
+
+# Mapping between config entry format and pyatv
+CREDENTIAL_MAPPING = {
+    CONF_CREDENTIALS_MRP: const.Protocol.MRP.value,
+    CONF_CREDENTIALS_DMAP: const.Protocol.DMAP.value,
+    CONF_CREDENTIALS_AIRPLAY: const.Protocol.AirPlay.value,
+}
+
+
+async def device_scan(identifier, loop, cache=None):
+    """Scan for a specific device using identifier as filter."""
+
+    def _filter_device(dev):
+        if identifier is None:
+            return True
+        if identifier == str(dev.address):
+            return True
+        if identifier == dev.name:
+            return True
```

Comment:
Bring the different conditions together and connect them with `or` since they are all returning `True`.

Reply:
I don't want to do that for two reasons:

* The code will not be clearer because of this. It looks repetitive now but it is very easy to see what is happening.
* We only have code coverage based on lines and not branches. So if all checks were on one line, one test would "hide" the other cases that are also interesting to test.

Reply:
Good arguments, you convinced me!